src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
+            }
+
+            $indexes = [];
+            if (! empty($indexesData[$tableName])) {
+                $indexes = $this->_getPortableTableIndexesList(
+                    $indexesData[$tableName],
+                    $tableName
+                );
+            }
+
+            $table = new Table($tableName, $columns, $indexes, [], $foreignKeys, []);
+            if (! empty($columnsData[$tableName][0]['TABLE_COMMENTS'])) {
+                $table->addOption('comment', $columnsData[$tableName][0]['TABLE_COMMENTS']);
+            }
+
+            $tables[] = $table;
+        }
+
+        return $tables;
+    }
+
+    /**
+     * Gets all column definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseColumns(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT C.OWNER,
+                 C.TABLE_NAME,
+                 C.COLUMN_NAME,
+                 C.DATA_TYPE,
+                 C.DATA_DEFAULT,
+                 C.DATA_PRECISION,
+                 C.DATA_SCALE,
+                 C.CHAR_LENGTH,
+                 C.DATA_LENGTH,
+                 C.NULLABLE,
+                 D.COMMENTS,
+                 E.COMMENTS TABLE_COMMENTS
+            FROM ALL_TAB_COLUMNS C
+       LEFT JOIN ALL_TAB_COMMENTS E ON E.OWNER = C.OWNER AND E.TABLE_NAME = C.TABLE_NAME
+       LEFT JOIN ALL_COL_COMMENTS D ON D.OWNER = C.OWNER AND D.TABLE_NAME = C.TABLE_NAME AND
+                 D.COLUMN_NAME = C.COLUMN_NAME
+SQL;
+
+        $conditions = [];
+        $params     = [];
+
+        $conditions[]    = 'C.OWNER = :OWNER';
+        $params['OWNER'] = $databaseName;
+
+        if ($tableName !== null) {
+            $conditions[]         = 'C.TABLE_NAME = :TABLE_NAME';
+            $params['TABLE_NAME'] = $tableName;
+        }
+
+        $sql .= ' WHERE ' . implode(' AND ', $conditions);
+        $sql .= <<<'SQL'
+        ORDER BY C.TABLE_NAME, C.COLUMN_ID
+SQL;
+
+        return $this->_conn->fetchAllAssociative($sql, $params);
+    }
+
+    /**
+     * Gets all indexes definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseIndexes(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT IND_COL.TABLE_NAME,
+                 IND_COL.INDEX_NAME AS NAME,
+                 IND.INDEX_TYPE AS TYPE,
+                 DECODE(IND.UNIQUENESS, 'NONUNIQUE', 0, 'UNIQUE', 1) AS IS_UNIQUE,
+                 IND_COL.COLUMN_NAME AS COLUMN_NAME,
+                 IND_COL.COLUMN_POSITION AS COLUMN_POS,
+                 CON.CONSTRAINT_TYPE AS IS_PRIMARY
+            FROM ALL_IND_COLUMNS IND_COL
+       LEFT JOIN ALL_INDEXES IND ON IND.OWNER = IND_COL.INDEX_OWNER AND IND.INDEX_NAME = IND_COL.INDEX_NAME
+       LEFT JOIN ALL_CONSTRAINTS CON ON  CON.OWNER = IND_COL.INDEX_OWNER AND CON.INDEX_NAME = IND_COL.INDEX_NAME
```

Comment:
There's a double space here which is redundant.

Reply:
fixed